src/modules/flight_mode_manager/FlightModeManager.cpp

```diff
@@ -228,56 +241,35 @@ void FlightModeManager::start_flight_task()
 			break;
 		}
 
-		task_failure = error != FlightTaskError::NoError;
+		matching_task_running = matching_task_running && !task_failure && (error == FlightTaskError::NoError);
+		task_failure = (error != FlightTaskError::NoError);
```

Comment:
True 💡 because if any previous matching task failed then `matching_task_running` is already `false`. I was going through different possibilities during implementation. But now it seems quite comprehensible for what it does 👍

Reply:
I changed it in all 3 cases: https://github.com/PX4/PX4-Autopilot/compare/4fb5b16237b84e58c3bf190afcf0c974ff6eb4ac..a0b93ec9bff41da9217ed0bba168e347f78071e2#diff-75545eaa2cbcbcced0da88552481e3ce674572b655022ed05e642ec07869eaf6R225